File: typescript/secret-handshake/secret-handshake.ts
const Wink = 0b00001;
const DoubleBlink = 0b00010;
const CloseYourEyes = 0b00100;
const Jump = 0b01000;
const Reverse = 0b10000;

const commands = {
  wink: Wink,
  "double blink": DoubleBlink,
  "close your eyes": CloseYourEyes,
  jump: Jump,
};

export default class Handshake {
  constructor(private input: number) {}

  private isEnabled = (mask: number) => (this.input & mask) > 0;

  public commands(): string[] {
    const cmds = Object.entries(commands)
      .filter(([, mask]) => this.isEnabled(mask))
      .map(([cmd]) => cmd);

    if (this.isEnabled(Reverse)) {
      return cmds.reverse();
    }

    return cmds;
  }
}

File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SecretHandshake
{
    private static readonly IDictionary<int, Func<IEnumerable<string>, IEnumerable<string>>> AllCommands =
        new Dictionary<int, Func<IEnumerable<string>, IEnumerable<string>>>
        {
            {1 << 0, x => x.Append("wink")},
            {1 << 1, x => x.Append("double blink")},
            {1 << 2, x => x.Append("close your eyes")},
            {1 << 3, x => x.Append("jump")},
            {1 << 4, x => x.Reverse()},
        };

    public static string[] Commands(int mask) => AllCommands
        .Aggregate(
            Enumerable.Empty<string>(),
            (state, cmd) => (mask & cmd.Key) != 0 ? cmd.Value.Invoke(state) : state)
        .ToArray();
}